test/scales/tickGeneratorsTests.ts

```diff
@@ -22,35 +23,62 @@ describe("Scales", () => {
         assert.deepEqual(ticks, [-7.6, -7.5, -5, -2.5, -2.2], "generated all ticks between lower and higher value");
       });
 
-      it("only returns the ends of the domain if interval is bigger than the domain", () => {
-        let start = 0.5, end = 10.01, interval = 11;
+      it("returns the ends of the domain if interval is NaN or bigger than the domain", () => {
+        let start = 0.5;
+        let end = 10.01;
         scale.domain([start, end]);
-        let ticks = Plottable.Scales.TickGenerators.intervalTickGenerator(interval)(scale);
-        assert.deepEqual(ticks, [0.5, 10.01], "no middle ticks were added");
+        let ticks = Plottable.Scales.TickGenerators.intervalTickGenerator(11)(scale);
+        assert.deepEqual(ticks, [0.5, 10.01], "no middle ticks were added when interval is 11");
+
+        ticks = Plottable.Scales.TickGenerators.intervalTickGenerator(Infinity)(scale);
+        assert.deepEqual(ticks, [0.5, 10.01], "no middle ticks were added when interval is Infinity");
+
+        ticks = Plottable.Scales.TickGenerators.intervalTickGenerator(NaN)(scale);
+        assert.deepEqual(ticks, [0.5, 10.01], "no middle ticks were added when interval is NaN");
+      });
+
+      it("works for Scales.ModifiedLog", () => {
+        let logScale = new Plottable.Scales.ModifiedLog(2);
+        logScale.domain([0, 100]);
+        let logTicks = Plottable.Scales.TickGenerators.intervalTickGenerator(40)(logScale);
+        assert.deepEqual(logTicks, [0, 40, 80, 100], "generates ticks for Scales.ModifiedLog");
       });
 
-      it("rejects negative values", () => {
+      it("rejects non-positive values", () => {
         (<any>assert).throws(() => Plottable.Scales.TickGenerators.intervalTickGenerator(0), Error, "interval must be positive number",
-          "interval must be positive number");
+          "interval cannot be 0");
         (<any>assert).throws(() => Plottable.Scales.TickGenerators.intervalTickGenerator(-2), Error, "interval must be positive number",
-          "interval must be positive number");
+          "interval cannot be negatvie");
+      });
+
+      it.skip("rejects non-QuantitativeScale<number> Scales", () => {
+        let categoryScale: any = new Plottable.Scales.Category();
+        (<any>assert).throws(() => Plottable.Scales.TickGenerators.intervalTickGenerator(1)(categoryScale), Error,
+          "scale needs to inherit from Scale.QuantitativeScale<number>", "Scales.Category is not a valid parameter for TickGenerators");
+
+        let timeScale: any = new Plottable.Scales.Time();
+        (<any>assert).throws(() => Plottable.Scales.TickGenerators.intervalTickGenerator(1)(timeScale), Error,
+          "scale needs to inherit from Scale.QuantitativeScale<number>", "Scales.Time is not a valid parameter for TickGenerators");
       });
+
     });
 
-    describe("Integer", () => {
+    describe("integerTickGenerator()", () => {
```

Comment:
Same as above about semantically talking about the test suite rather than through endpoints